### PR TITLE
Add RECEIVER_EXPORTED flag to portrait sys ui register receiver calls;

### PR DESCRIPTION
**Pull request by VitaliDim**
Tested locally

- - -
This change is [Reviewable](https://reviewable.io/reviews/google/mobly-bundled-snippets/176)

### CODE REVIEW

**Comment by VitaliDim** [comment]
@xpconanfan , @ko1in1u , @xianyuanjia  please review this PR.
Not able to add reviews via reviewers input

**Comment by xpconanfan** [comment]
https://developer.android.com/reference/android/content/Context#RECEIVER_EXPORTED

This is added in API level 33.

What happens when users try to use this on older Android versions?

**Review by xpconanfan — CHANGES_REQUESTED** [review]
Don't we need to handle compatibility with lower API levels?

**Comment by VitaliDim** [comment]
Done, @xpconanfan please take a look.
Thank you

**Review by xianyuanjia — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/google/mobly-bundled-snippets/176#-:-NdaFE1y1savoH1_7qLM:bnfp4nl)* status: 0 of 1 files reviewed, all discussions resolved (waiting on @xpconanfan)

**Comment by VitaliDim** [comment]
RPC verified, works as expected.
Tested on build: 10651996, target: seahawk-userdebug, SDK: 34

**Comment by xpconanfan** [comment]
> fied, works as expected.
> Tested on build: 10651996, target: seahaw

Thanks for this.
But the main concern is this breaking for lower level APIs.
Did you test on lower level APIs and phones without changing the build config?

**Comment by VitaliDim** [comment]
@xpconanfan I did not tested on lover APIs level.
For now we need move forward with BT Messenger automation.
This changes its a workaround for unblocking our efforts for SDK 34 and above.

**Comment by VitaliDim** [comment]
@xpconanfan code for lower version SDK has not been changed at all. The changes related only for version 33 and above.

**Comment by VitaliDim** [comment]
@xpconanfan verified on SDK 31, works as expected

File: src/main/java/com/google/android/mobly/snippet/bundled/SmsSnippet.java [hunk 1]
```diff
@@ -107,7 +107,13 @@ public void sendSms(String phoneNumber, String message) throws Throwable {
                             /* intent= */ new Intent(SMS_SENT_ACTION),
                             /* flags= */ PendingIntent.FLAG_IMMUTABLE);
             receiver.setExpectedMessageCount(1);
-            mContext.registerReceiver(receiver, new IntentFilter(SMS_SENT_ACTION));
+            if (Build.VERSION.SDK_INT >= 33) {
```
Comment (ko1in1u) [inline src/main/java/com/google/android/mobly/snippet/bundled/SmsSnippet.java:110]: looks like we need the same change at line 87

**Review by xpconanfan — APPROVED** [review]
Leaving the final Approval to [ko1in1u](https://github.com/ko1in1u)

**Review by ko1in1u — APPROVED** [review]

**Review by VitaliDim — COMMENTED** [review]
> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/google/mobly-bundled-snippets/176)* status: all files reviewed, 1 unresolved discussion (waiting on @VitaliDim)

**Pull request merged**